pyqtgraph/graphicsItems/ErrorBarItem.py

```diff
@@ -46,10 +46,10 @@ def setData(self, **opts):
         This method was added in version 0.9.9. For prior versions, use setOpts.
         """
         self.opts.update(opts)
-        if self.opts['x'] is not None and self.opts['y'] is not None:
-            self.setVisible(True)
-        else:
+        if self.opts['x'] is None or self.opts['y'] is None:
```

Comment:
Don't meant to get nitpickey; but should we not do something like

```
self.setVisible(True) if all(self.opts[axis] is not None for axis in {'x', 'y'}) else set.Visible(False)
```

Reply:
Isn't that equivalent? Or are you just talking stylistically? While we're at it, this is pretty concise:

```python
self.setVisible(all(self.opts[ax] is not None for ax in {'x', 'y'}))
```

Reply:
oh yeah these are equivalent; I've recently been on a ternary operator kick ...and I'm seeing many opportunities to use them and don't ...

Also, if there is a more likely axis to have `self.opts[axis] is None`, we should likely specify the order in list form, instead of the set notation I used.

Reply:
I like your method even more

Reply:
The check in `drawPath` is `if x is None or y is None`, which will short-circuit if x is None. Similarly, if x is None here, the second compare won't happen if we specify `all(self.opts[ax] for ax in ['x', 'y'])` :+1:

Reply:
Just to have it mentioned: Readability should be more important than line count, as it directly affects how good Pull Requests from less experienced users are. - Having said that, I also really like to play Code Golf myself ^^

Reply:
> Just to have it mentioned: Readability should be more important than line count, as it directly affects how good Pull Requests from less experienced users are. - Having said that, I also really like to play Code Golf myself ^^

Oh I'm totally down for readability; but an `if/elif` check in this case where was pretty verbose, and while capturing the necessary logic, I do think using the `all` keyword is appropriate here.

Then again... https://i.imgur.com/ZgQHMHo.png